modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/cache/ClientCacheQueryContinuousRequest.java

```diff
@@ -0,0 +1,147 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.platform.client.cache;
+
+import javax.cache.configuration.Factory;
+import javax.cache.configuration.FactoryBuilder;
+import javax.cache.event.CacheEntryEventFilter;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.binary.BinaryObject;
+import org.apache.ignite.cache.CacheEntryEventSerializableFilter;
+import org.apache.ignite.cache.query.ContinuousQuery;
+import org.apache.ignite.cache.query.QueryCursor;
+import org.apache.ignite.internal.binary.BinaryObjectImpl;
+import org.apache.ignite.internal.binary.BinaryRawReaderEx;
+import org.apache.ignite.internal.binary.GridBinaryMarshaller;
+import org.apache.ignite.internal.processors.platform.PlatformContext;
+import org.apache.ignite.internal.processors.platform.PlatformJavaObjectFactoryProxy;
+import org.apache.ignite.internal.processors.platform.client.ClientConnectionContext;
+import org.apache.ignite.internal.processors.platform.client.ClientPlatform;
+import org.apache.ignite.internal.processors.platform.client.ClientResponse;
+import org.apache.ignite.internal.processors.platform.client.ClientStatus;
+import org.apache.ignite.internal.processors.platform.client.IgniteClientException;
+import org.apache.ignite.internal.processors.platform.utils.PlatformUtils;
+
+/**
+ * Continuous query request.
+ */
+@SuppressWarnings({"rawtypes", "unchecked"})
+public class ClientCacheQueryContinuousRequest extends ClientCacheRequest {
+    /** Query. */
+    private final ContinuousQuery qry;
+
+    /** */
+    private final Object filter;
+
+    /** */
+    private final byte filterPlatform;
+
+    /**
+     * Ctor.
+     *
+     * @param reader Reader.
+     */
+    public ClientCacheQueryContinuousRequest(BinaryRawReaderEx reader) {
+        super(reader);
+
+        int pageSize = reader.readInt();
+        long timeInterval = reader.readLong();
+        boolean includeExpired = reader.readBoolean();
+
+        filter = reader.readObjectDetached();
+        filterPlatform = filter == null ? 0 : reader.readByte();
+
+        qry = new ContinuousQuery();
+
+        qry.setPageSize(pageSize)
+           .setTimeInterval(timeInterval)
+           .setIncludeExpired(includeExpired);
+    }
+
+    /** {@inheritDoc} */
+    @Override public ClientResponse process(ClientConnectionContext ctx) {
+        qry.setRemoteFilterFactory(getFilterFactory(ctx));
+
+        ctx.incrementCursors();
```

Comment:
Are you sure we should treat it as a query? I know there is a query word in name, but I'm not sure those mechanisms belong to the same pool.

Reply:
Good point, I've posted this to the dev list.

Reply:
No replies on dev list. Let's keep it as is for now. Continuous Query still gives us a `QueryCursor`, so I think it fits with the other cursors.